src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs

```diff
@@ -466,6 +469,71 @@ RequestContext<object> requestContext
             await requestContext.SendResult(result.First().ToString());
         }
 
+        private async Task HandleGetAllCommandsRequest(
+            object param,
+            RequestContext<object> requestContext)
+        {
+            var psCommand = new PSCommand();
+            psCommand.AddScript("Get-Command -CommandType Function, Cmdlet, ExternalScript | Select-Object Name,ModuleName | Sort-Object Name");
+            Logger.Write(LogLevel.Verbose, $"Calling {psCommand.Commands}");
+            Stopwatch stopwatch = new Stopwatch();
+            stopwatch.Start();
+            var result = await this.editorSession.PowerShellContext.ExecuteCommand<PSObject>(psCommand);
+            stopwatch.Stop();
+            Logger.Write(LogLevel.Verbose, $"Returned from Get-Command with {result.Count()} results in {stopwatch.ElapsedMilliseconds}ms");
+            var commandList = new List<PSAllCommandsMessage>();
```

Comment:
If we already call count (and the result is secretly a `Collection<T>` anyway), this list can be initialised with the number of objects expected. Or it could be an array